packages/schema-blocks/src/functions/validators/innerBlocksValid.ts

```diff
@@ -78,46 +84,59 @@ function validateBlocks( requiredBlocks: RequiredBlock[], blocks: BlockInstance[
 /**
  * Validates all inner blocks recursively and returns all invalid blocks.
  *
+ * @param blockInstance  The block whose inner blocks need to be validated.
  * @param requiredBlocks Requirements of the blocks that should occur in the inner blocks.
- * @param clientId       The clientId of the block whose inner blocks need to be validated.
  *
  * @returns {InvalidBlock[]} The names and reasons of the inner block that are invalid.
  */
-function getInvalidInnerBlocks( requiredBlocks: RequiredBlock[], clientId: string ): InvalidBlock[]  {
+function getInvalidInnerBlocks( blockInstance: BlockInstance, requiredBlocks: RequiredBlock[] ): InvalidBlock[]  {
+	const requiredBlockKeys = requiredBlocks.map( rblock => rblock.name );
 	const invalidBlocks: InvalidBlock[] = [];
 
-	const innerBlocks = getInnerBlocks( clientId );
-	const requiredBlockKeys = Object.keys( requiredBlocks );
-
 	// Find all instances of required block types.
-	const existingRequiredBlocks = getInnerblocksByName( requiredBlockKeys, innerBlocks );
+	const existingRequiredBlocks = getInnerblocksByName( blockInstance, requiredBlockKeys );
 
 	// Find all block types that do not occur in existingBlocks.
-	invalidBlocks.push( ...findMissingBlocks( requiredBlocks, existingRequiredBlocks ) );
+	invalidBlocks.push( ...findMissingBlocks( existingRequiredBlocks, requiredBlocks ) );
 
 	// Find all block types that allow only one occurrence.
-	invalidBlocks.push( ...findRedundantBlocks( requiredBlocks, existingRequiredBlocks ) );
+	invalidBlocks.push( ...findRedundantBlocks( existingRequiredBlocks, requiredBlocks ) );
 
 	// Find all blocks that have decided for themselves that they're invalid.
-	invalidBlocks.push( ...validateBlocks( requiredBlocks, innerBlocks ) );
+	// InvalidBlocks.push( ...findSelfInvalidatedBlocks( blockInstance, requiredBlocks ) );
 
 	return invalidBlocks;
 }
 
 /**
  * Helper function to create an invalid block.
  *
- * @param type   The block name.
+ * @param name   The block name.
  * @param reason The reason this block is invalid.
  *
  * @returns {InvalidBlock} The invalid block.
  */
-function createInvalidBlock( type: string, reason: InvalidBlockReason ): InvalidBlock {
-	return {
-		name: type,
+function createInvalidBlock( name: string, reason: InvalidBlockReason ): InvalidBlock {
+	const block: InvalidBlock = {
+		name: name,
 		reason,
-	} as InvalidBlock;
+	};
+	return block;
+}
+
+/**
+ * Helper function to determine the urgency of an invalidation.
+ *
+ * @param reason The reason to check.
+ *
+ * @returns {boolean} True if the invalidation is for an optional block (a warning), false if the invalidation is for a required block (an error).
+ */
+function isOptional( reason: InvalidBlockReason ): boolean {
+	switch ( reason ) {
+		case InvalidBlockReason.Optional: return true;
+		default: return false;
+	}
```

Comment:
Please use some newlines to make this more readable
```suggestion
	switch ( reason ) {
		case InvalidBlockReason.Optional: 
			return true;
		default: 
			return false;
	}
```